stix_shifter_modules/vision_one/stix_transmission/api_client.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+import json
+
+from stix_shifter_utils.stix_transmission.utils.RestApiClient import RestApiClient
+from stix_shifter_utils.utils import logger
+
+_USER_AGENT = 'TMV1StixShifter/1.0'
+
+
+class APIClient:
+
+    def __init__(self, connection, configuration):
+        self.logger = logger.set_logger(__name__)
+        self.endpoint_start = 'v2.0/xdr/'
+        headers = dict()
+        host_port = connection.get('host') + ':' + str(connection.get('port', ''))
+        headers['accept'] = 'application/json'
+        auth = configuration.get('auth')
+        if auth is not None and auth.get('tok', None) is not None:
+            headers['Authorization'] = 'Bearer {}'.format(auth.get('tok'))
+        url_modifier_function = None
+        headers['user-agent'] = _USER_AGENT
+
+        self.timeout = connection['options'].get('timeout')
+
+        self.client = RestApiClient(host_port,
+                                    None,
+                                    headers,
+                                    url_modifier_function,
+                                    cert_verify=connection.get('selfSignedCert', True),
+                                    sni=connection.get('sni', None)
+                                    )
+
+    def ping_data_source(self):
+        # Pings the data source
+        endpoint = 'v2.0/siem/events'
+        now = datetime.datetime.utcnow().isoformat(timespec="milliseconds") + "Z"
+        resp = self.client.call_api(endpoint, 'GET', timeout=self.timeout, urldata={"startDateTime": now, "endDateTime": now})
+        return {"code": resp.code}
```

Comment:
why are you just returning the response code? PingConnector class calls this method to get the response. response is already handled in `PingConnector.ping_connection()`

Reply:
This API is just to check whether the target server is available and token is valid.